UMLEditor/UMLEditor/Views/RelationshipLine.cs

```diff
@@ -70,33 +77,89 @@ public RelationshipLine(UserControl startCtrl, UserControl endCtrl, string relat
         Segments = new List<Line>();
     }
 
+    /// <summary>
+    /// Initializes the pathing grid to the correct size and creates nodes for each coordinate
+    /// </summary>
+    /// <param name="canvas">The canvas to get size of for grid (currently not working)</param>
+    public static void InitializeGrid(Canvas canvas)
+    {
+        // Currently not getting bounds correctly, using placeholder values
+        int maxX = 2000 / Node.NODE_SIZE, maxY = 2000 / Node.NODE_SIZE;
+        if (canvas.Bounds.Width != 0)
+        {
+            maxX = (int)canvas.Bounds.Width;
+        }
+
+        if (canvas.Bounds.Height != 0)
+        {
+            maxY = (int)canvas.Bounds.Height;
+        }
+        
+        for (int x = 0; x < maxX; ++x)
+        {
+            List<Node> row = new List<Node>();
+            _grid.Add(row);
+            for (int y = 0; y < maxY; ++y)
+            {
+                Node myNode = new Node(new Vector2(x,y), true);
+                _grid[x].Add(myNode);
+            }
+        }
+    }
+
+    /// <summary>
+    /// Resets the pathing grid to be fully walkable
+    /// </summary>
+    public static void ResetGrid()
+    {
+        for (int x = 0; x < _grid.Count; ++x)
+        {
+            for (int y = 0; y < _grid[x].Count; ++y)
+            {
+                _grid[x][y].Walkable = true;
+            }
+        }
+    }
+    
+    /// <summary>
+    /// Gets the straight line distance between two points
+    /// </summary>
+    /// <param name="p1">First point</param>
+    /// <param name="p2">Second point</param>
+    /// <returns></returns>
     private double GetDistance(Point p1, Point p2)
     {
         return Math.Sqrt(Math.Pow(p1.X - p2.X, 2) + Math.Pow(p1.Y - p2.Y,2));
     }
 
+    /// <summary>
+    /// Sets a grid location to not be walkable for the path search algorithm
+    /// </summary>
+    /// <param name="x">X coordinate of point</param>
+    /// <param name="y">Y coordinate of point</param>
+    public static void MakeNotWalkable(int x, int y)
+    {
+        _grid[x][y].Walkable = false;
+    }
+
     /// <summary>
     /// Function to draw the line.
     /// </summary>
     /// <param name="myCanvas">The canvas in which the line will be drawn.</param>
     public void Draw(Canvas myCanvas)
     {
+        // Get ClassBox objects for source and destination
         ClassBox sourceClassBox = ClassBoxes.FindByName(SourceClass.Name);
```

Comment:
![image](https://user-images.githubusercontent.com/89474048/164269844-294687be-2442-4d50-875b-0e247c80bac4.png)
We should *not* be using the name property of the control. This is unreliable and is not guaranteed to be updated when a class is renamed (we had this issue in an earlier sprint). The source and destinations should be classboxes and the ClassName property should be used for getting the box's name. Right now this causes the program to crash whenever a ClassBox is renamed.